### PR TITLE
[CuTe, SM90] Fix swapped backward MMA layout and use 2-stage hdim256 config

### LINKED ISSUE #2878: [FA4, Hopper, BWD] BWD Speed Regression on H200 when headdim=256
flash-attn-4    version:      4.0.0b24

Supposely, the runtime of bwd of FA4 should be nearly 500ms given the runtime of headdim=128. Unfortunately, it takes almost twice the runtime. 

```
batch_size: 1,  seq_len: 86000, head_size: 32, dim: 128
runtime of fwd of FA4 at bf16 precision in milliseconds: 88.88246 (88.88246, 88.88246).
---------------------------------------------------------------------
runtime of bwd of FA4 at bf16 precision in milliseconds: 246.29193 (246.29193, 246.29193).
---------------------------------------------------------------------
batch_size: 1,  seq_len: 86000, head_size: 32, dim: 256
runtime of fwd of FA4 at bf16 precision in milliseconds: 166.24957 (166.24957, 166.24957).
---------------------------------------------------------------------
runtime of bwd of FA4 at bf16 precision in milliseconds: 965.84863 (965.84863, 965.84863).
---------------------------------------------------------------------
```

To reproduce, run the following code on H200 under 4.0.0b24:

```
import torch
import triton
from flash_attn.cute.interface import flash_attn_varlen_func as FA4_Varlen

def benchmark(f, job_name:str):
    warmup=20
    rep=100
    ms = triton.testing.do_bench(f, warmup=warmup, rep=rep, quantiles=[0.2, 0.5, 0.8])
    print(f"runtime of {job_name} in milliseconds: {ms[1]:.5f} ({ms[0]:.5f}, {ms[2]:.5f}).")
    return ms[1]

def test_once_varlen(batch_size, seq_len, num_heads, dim):
    print(f"batch_size: {batch_size},  seq_len: {seq_len}, head_size: {num_heads}, dim: {dim}")
    q, k, v = get_tensors(batch_size, seq_len, num_heads, dim)
    device = "cuda"
    cu_q = torch.tensor([0, seq_len >> 1, seq_len], device=device, dtype=torch.int32)
    cu_k = torch.tensor([0, seq_len >> 1, seq_len], device=device, dtype=torch.int32)
    max_seqlen_q = int((cu_q[1:] - cu_q[:-1]).max())
    max_seqlen_k = int((cu_k[1:] - cu_k[:-1]).max())
    fa4_fwd_bf16 = lambda: FA4_Varlen(
        q[0], k[0], v[0],
        cu_seqlens_q=cu_q, cu_seqlens_k=cu_k,
        max_seqlen_q=max_seqlen_q, max_seqlen_k=max_seqlen_k,
    )
    runtime_fa4_fwd_bf16 = benchmark(fa4_fwd_bf16, "fwd of FA4 at bf16 precision")
    print("---------------------------------------------------------------------")

    out, _ = fa4_fwd_bf16()
    dout = torch.randn_like(out)
    fa4_bwd_bf16 = lambda: out.backward(dout, retain_graph=True)
    runtime_fa4_bwd_bf16 = benchmark(fa4_bwd_bf16, "bwd of FA4 at bf16 precision")
    print("---------------------------------------------------------------------")

    return runtime_fa4_fwd_bf16, runtime_fa4_bwd_bf16

def get_tensors(batch_size, seq_len, head_size, dim):
    torch.manual_seed(42)
    q = torch.randn((batch_size, seq_len, head_size, dim), dtype=torch.bfloat16, device="cuda", requires_grad=True)
    k = torch.randn((batch_size, seq_len, head_size, dim), dtype=torch.bfloat16, device="cuda", requires_grad=True)
    v = torch.randn((batch_size, seq_len, head_size, dim), dtype=torch.bfloat16, device="cuda", requires_grad=True)
    return q, k, v

if __name__ == "__main__":
    test_once_varlen(1, 86000, 32, 128)
    test_once_varlen(1, 86000, 32, 256)
```

### PR DESCRIPTION
**Pull request by SuperGoodGame**
## Summary

SM90 backward at head_dim 256 runs at ~300 TFLOP/s while hdim 128 is at parity with the C++ FA3 kernel (#2878). The cause is the hdim 256 config: it is the only SM90 backward config with `num_stages_Q=1`, so the Q/dO TMA loads and the WGMMAs are serialized every m-block. The C++ tile (64x80, 2-stage Q) does not fit next to the 64 KB smem dQaccum staging buffer, but 64x48 with `dKV_swapAB` does.

Using that config for GQA/MQA as well needs one real fix, and it turned up a second one:

1. **Swapped dK/dV postprocess layout.** `FlashAttentionBackwardPostprocess._get_tiled_mma` reversed the whole `(tile / atom_layout)` tiler for `swap_ab`, which asks `make_trivial_tiled_mma` for a WGMMA atom with M = hdim / num_wg. Every SM90 config with `dKV_swapAB` therefore fails in the GQA/MQA postprocess: `expects the M-mode to be 64, but got 192` for hdim in (128, 192] on `main` today (cf. #2087), `got 128` for `dQ_swapAB` above hdim 128. Build the tiled MMA the way the mainloop does, `tiler_mn=(64, tile / atom_layout on N)`; the accumulator register layout is unchanged.
2. **GQA dV accumulator rounding.** `dv_accum` was allocated with a fixed 32-element head-dim rounding while the kernel and the postprocess use `hdim_multiple_of` (64 with `dKV_swapAB`), so head_dim_v in {160, 200, 208, 224} reduced and read past each (batch, kv-head) region and dV came out wrong. Round it like `dk_accum`.
3. **hdim 256 config: 64x48, 2-stage Q, `dKV_swapAB`**, the same for MHA, GQA and MQA (no per-mode fallback). `dKV_swapAB` rounds `tile_hdim` to 64, so head_dim in (192, 256) (200/208/224/240, which currently abort inside the DSL compiler or fail in `get_smem_store_C`) now compiles and is correct. Block-sparse backward keeps the previous 64x64 tile because it derives its KV block size from `n_block_size`.

No new kernel or config parameters; forward and hdim <= 192 configs are untouched. 3 files, +40/-9 including tests.

## Results

H20Z (H200 silicon, 132 SMs, 227 KB smem; CUDA 12.8, torch 2.10, nvidia-cutlass-dsl 4.7.1), bf16, backward:

| | `main` (8d3a3b8) | this PR |
|---|---|---|
| dense S=16384 H=16 D=256 | 36.5 ms · 302 TFLOP/s | **29.5 ms · 373** |
| dense causal S=16384 H=16 D=256 | 17.2 ms · 320 | **14.8 ms · 370** |
| varlen S=86000 H=32 D=256 (the shape in #2878) | 1931 ms · 314 | **1672 ms · 362** |
| varlen causal S=86000 H=32 D=256 | — | 790 ms · 383 |
| GQA hdim 192, hdim 200/208/224/240 | fail | work |

The remaining gap to C++ FA3 (~480 TFLOP/s here) is the smem dQaccum staging; that is a separate, larger change (C++ FA3's `dQacc_use_TMA=false` path) and will be a follow-up PR.

## Correctness

- Max error vs an fp32 reference (S=2048) is unchanged from `main`: dq/dk/dv 1.33e-3 / 1.16e-3 / 9.14e-4 non-causal, 1.20e-2 / 1.21e-2 / 1.58e-2 causal (identical numbers on `main`).
- Checked vs the fp32 reference: dense, causal, GQA 16/4, GQA causal 16/2, MQA, local (200,0) and (100,50), softcap, deterministic (bitwise-identical repeat, MHA and GQA), hdim_v=128, hdim 200/224/240 for MHA and GQA (per kv head), ragged S=1000, S=4096.
- `tests/cute/test_flash_attn.py::test_flash_attn_output`, d=256 subset (seqlen pairs 1-1 / 128-128 / 113-203 / 256-512 / 384-256 / 799-3 × causal × local_enum × softcap × deterministic × learnable_sink × mha/mqa/gqa, 1152 ids): **720 passed, 432 skipped (local+causal combinations the test skips), 0 failed**.
- `test_flash_attn_sm90_hdim_padding*` (this PR extends `d` to 200/224/240 and adds a GQA variant for d = 160/192/224): **150 passed** (was 96 cases on `main`, where the new ones fail).
- `ruff check` / `ruff format --check` (v0.11.13, as in pre-commit) clean on the CI-covered file.

Fixes #2878. Related: #2087.

### CODE REVIEW

**Review by drisspg — APPROVED** [review]
verified; and this fixes thanks!
